napari/_qt/widgets/qt_dims_sorter.py

```diff
@@ -64,37 +51,56 @@ def move_indices(axes_list: SelectableEventedList, order: Tuple[int, ...]):
 
 
 class QtDimsSorter(QWidget):
-    """
+    """Qt widget for dimension / axis reordering.
+
     Modified from:
     https://github.com/jni/zarpaint/blob/main/zarpaint/_dims_chooser.py
+
+    Parameters
+    ----------
+    viewer : napari.Viewer
+        Main napari viewer instance.
+    parent : QWidget, optional
+        QWidget that will be the parent of this widget. 
+    
+    Attributes
+    ----------
+    axis_list : napari._qt.containers.qt_axis_model.AxisList
+        Selectable evented list representing the viewer axes.
     """
 
     def __init__(self, viewer: 'Viewer', parent=None) -> None:
         super().__init__(parent=parent)
         dims = viewer.dims
-        root = SelectableEventedList(
-            [AxisModel(dims, dims.order[i]) for i in range(dims.ndim)]
-        )
-        root.events.reordered.connect(
-            lambda event: set_dims_order(dims, event.value)
-        )
-        dims.events.order.connect(
-            lambda event: move_indices(root, event.value)
-        )
-        view = QtListView(root)
-        view.setSizeAdjustPolicy(QtListView.AdjustToContents)
+        self.axis_list = AxisList.from_dims(dims)
 
-        self.axes_list = root
+        view = QtListView(self.axis_list)
+        view.setSizeAdjustPolicy(QtListView.AdjustToContents)
 
         layout = QGridLayout()
         self.setLayout(layout)
 
         widget_tooltip = QtToolTipLabel(self)
         widget_tooltip.setObjectName('help_label')
-        widget_tooltip.setToolTip(trans._('Drag dimensions to reorder.'))
+        widget_tooltip.setToolTip(
+            trans._(
+                'Drag dimensions to reorder, uncheck to lock dimension in place.'
+            )
+        )
 
         widget_title = QLabel(trans._('Dims. Ordering'), self)
 
         self.layout().addWidget(widget_title, 0, 0)
         self.layout().addWidget(widget_tooltip, 0, 1)
         self.layout().addWidget(view, 1, 0)
```

Comment:
Not related to your changes, but since I became aware of this widget, it's layout started bothering me :D 

One simple, small improvement is to make the list span the two columns of the grid layout.

```suggestion
        self.layout().addWidget(view, 1, 0, 1, 2)
```

Generally, the layout and styling could do with a rethink. But this small improvement is worth it.